### PR TITLE
Add keyboard: Mesa TKL

### PR DESCRIPTION
**Pull request by MesaKeyboards**
## Description

Adds a new keyboard: Mesa TKL

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mesa/mesa_tkl/mesa_tkl.c [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// Unused
```
Comment (zvecr) [inline keyboards/mesa/mesa_tkl/mesa_tkl.c:17]:
```suggestion
#include "mesa_tkl.h"
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/mesa/mesa_tkl/config.h [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID	0x4E5A
+#define PRODUCT_ID	0x8001
+#define DEVICE_VER	0x0001
+#define MANUFACTURER	Mesa Keyboards
+#define PRODUCT		Mesa TKL
```
Comment (fauxpark) [inline keyboards/mesa/mesa_tkl/config.h:27]:
```suggestion
#define MANUFACTURER Mesa Keyboards
#define PRODUCT      Mesa TKL
```

File: keyboards/mesa/mesa_tkl/rules.mk [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+CONSOLE_ENABLE    = no        # Console for debug
+COMMAND_ENABLE    = no        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE  = no        # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE       = yes       # USB Nkey Rollover
+BACKLIGHT_ENABLE  = no        # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE   = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE  = no        # Enable Bluetooth
+AUDIO_ENABLE      = no        # Audio output
```
Comment (fauxpark) [inline keyboards/mesa/mesa_tkl/rules.mk:22]:
```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mesa/mesa_tkl/config.h (hunk 2 above)
Comment (MesaKeyboards) [reply keyboards/mesa/mesa_tkl/config.h]: I've replaced all tabs with spaces in e6fd58.

**Review by drashna — APPROVED** [review]

File: keyboards/mesa/mesa_tkl/readme.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# Mesa TKL
+
+![Mesa TKL](https://imgur.com/XPLJmr8.png)
```
Comment (fauxpark) [inline keyboards/mesa/mesa_tkl/readme.md:3]:
```suggestion
![Mesa TKL](https://imgur.com/XPLJmr8l.png)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by MesaKeyboards** [comment]
@zvecr just checking in. I committed your requested change from a few weeks ago - is there anything else you need from me?

**Pull request merged**